readthedocs/core/views/hooks.py

```diff
@@ -165,14 +164,21 @@ def deactivate_external_version(project, identifier, verbose_name):
     so another celery task will remove it after some days.
 
     :param project: Project instance
+    :param version_data: A :py:class:`readthedocs.api.v2.views.integrations.ExternalVersionData`
+                         instance.
     :param identifier: Commit Hash
     :param verbose_name: pull/merge request number
     :returns: verbose_name (pull/merge request number).
     :rtype: str
     """
-    external_version = project.versions(manager=EXTERNAL).filter(
-        verbose_name=verbose_name, identifier=identifier
-    ).first()
+    external_version = (
+        project.versions(manager=EXTERNAL)
+        .filter(
+            verbose_name=version_data.id,
+            identifier=version_data.commit,
```

Comment:
Feels weird to have these names mismatched on the model & the object. Are the names in `version_data` coming from an external source, and we're just trying to map them onto our Version?

Reply:
Yeah, that's the case.